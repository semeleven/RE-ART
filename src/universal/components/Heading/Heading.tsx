import React from 'react';
import styledProps from 'styled-props';

import styled from '../../lib/styled';
import { fontSize, colors, headingMargin } from '../../lib/styled/theme';

interface Props {
	size?: 'XL' | 'L' | 'M' | 'S';
	mono?: boolean;
	children: React.ReactNode;
	className?: string;
	white?: boolean
	black?: boolean
	lighterGray?: boolean
	uppercase?: boolean
	marginTop?: number | string
	marginBottom?: number | string
}

const Sizes = {
	XL: 1,
	L: 2,
	M: 3,
	S: 4,
};

const Heading: React.SFC<Props> = ({ size = 'XL', className, children }) =>
	React.createElement(`h${Sizes[size]}`, { className }, children);

const StyledHeading = styled(Heading)`
	font-size: ${props => fontSize[props.size]};
	font-family: ${props =>
		(props.mono ? 'Source Code Pro, monospace' : 'Lato, sans-serif')};
	color: ${styledProps(colors)};
	text-transform: ${props => props.uppercase && 'uppercase'};
	margin-top: ${props =>
		props.marginTop
			? props.marginTop + 'px'
			: headingMargin[props.size]
	};
	margin-bottom: ${props => 
		props.marginBottom 
			? props.marginBottom + 'px' 
			: headingMargin[props.size]
	};
`;

StyledHeading.defaultProps = {
	white: true,
	size: 'XL',
	mono: false,
	uppercase: false,
	marginTop: null,
	marginBottom: null,
};

export default StyledHeading;
