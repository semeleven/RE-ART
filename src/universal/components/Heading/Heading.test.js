import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Heading from './Heading';
import sinon from 'sinon';

const sizes = ['XL', 'L', 'M', 'S'];

for (let i = 0; i < sizes.length; i++) {
	const currentSize = sizes[i];

	it(`renders black ${currentSize} Heading`, () => {
		const wrapper = shallow(
			<Heading black size={currentSize}>
				${currentSize}
			</Heading>
		);
		expect(wrapper).toMatchSnapshot();
	});
}

it(`renders mono dark gray Heading with default prop XL`, () => {
	const wrapper = shallow(
		<Heading mono darkGray>
			DARK GRAY MONO
		</Heading>
	);
	expect(wrapper).toMatchSnapshot();
});

it(`renders bold and uppercase lighter gray Heading`, () => {
	const wrapper = shallow(
		<Heading bold uppercase lighterGray>
			MONO
		</Heading>
	);
	expect(wrapper).toMatchSnapshot();
});

it(`renders black (default color prop) Heading with vertical margins`, () => {
	const wrapper = shallow(
		<Heading marginBottom="30px" marginTop="30px">
			MONO
		</Heading>
	);
	expect(wrapper).toMatchSnapshot();
});

it(`renders purple Heading`, () => {
	const wrapper = shallow(<Heading purple>MONO</Heading>);
	expect(wrapper).toMatchSnapshot();
});

it(`renders dark purple Heading`, () => {
	const wrapper = shallow(<Heading darkPurple>MONO</Heading>);
	expect(wrapper).toMatchSnapshot();
});

it('simulates click events', () => {
	const onHeadingClick = sinon.spy();
	const wrapper = shallow(<Heading onClick={onHeadingClick}>CLICK!</Heading>);
	wrapper.find('Heading').simulate('click');
	expect(onHeadingClick.calledOnce).toEqual(true);
});
