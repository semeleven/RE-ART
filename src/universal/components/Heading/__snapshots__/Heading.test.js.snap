// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders black (default color prop) Heading with vertical margins 1`] = `
.c0 {
  font-size: 40px;
  font-family: Lato,sans-serif;
  margin-top: 30px;
  margin-bottom: 30px;
}

<Heading
  bold={false}
  className="c0"
  marginBottom="30px"
  marginTop="30px"
  mono={false}
  size="XL"
  uppercase={false}
>
  MONO
</Heading>
`;

exports[`renders black L Heading 1`] = `
.c0 {
  font-size: 30px;
  font-family: Lato,sans-serif;
  color: #000000;
  margin-top: 40px;
  margin-bottom: 40px;
}

<Heading
  black={true}
  bold={false}
  className="c0"
  marginBottom={null}
  marginTop={null}
  mono={false}
  size="L"
  uppercase={false}
>
  $
  L
</Heading>
`;

exports[`renders black M Heading 1`] = `
.c0 {
  font-size: 17px;
  font-family: Lato,sans-serif;
  color: #000000;
  margin-top: 20px;
  margin-bottom: 20px;
}

<Heading
  black={true}
  bold={false}
  className="c0"
  marginBottom={null}
  marginTop={null}
  mono={false}
  size="M"
  uppercase={false}
>
  $
  M
</Heading>
`;

exports[`renders black S Heading 1`] = `
.c0 {
  font-size: 15px;
  font-family: Lato,sans-serif;
  color: #000000;
  margin-top: 10px;
  margin-bottom: 10px;
}

<Heading
  black={true}
  bold={false}
  className="c0"
  marginBottom={null}
  marginTop={null}
  mono={false}
  size="S"
  uppercase={false}
>
  $
  S
</Heading>
`;

exports[`renders black XL Heading 1`] = `
.c0 {
  font-size: 40px;
  font-family: Lato,sans-serif;
  color: #000000;
  margin-top: 50px;
  margin-bottom: 50px;
}

<Heading
  black={true}
  bold={false}
  className="c0"
  marginBottom={null}
  marginTop={null}
  mono={false}
  size="XL"
  uppercase={false}
>
  $
  XL
</Heading>
`;

exports[`renders bold and uppercase lighter gray Heading 1`] = `
.c0 {
  font-size: 40px;
  font-family: Lato,sans-serif;
  font-weight: bold;
  color: #707070;
  text-transform: uppercase;
  margin-top: 50px;
  margin-bottom: 50px;
}

<Heading
  bold={true}
  className="c0"
  lighterGray={true}
  marginBottom={null}
  marginTop={null}
  mono={false}
  size="XL"
  uppercase={true}
>
  MONO
</Heading>
`;

exports[`renders dark purple Heading 1`] = `
.c0 {
  font-size: 40px;
  font-family: Lato,sans-serif;
  color: #5346AE;
  margin-top: 50px;
  margin-bottom: 50px;
}

<Heading
  bold={false}
  className="c0"
  darkPurple={true}
  marginBottom={null}
  marginTop={null}
  mono={false}
  size="XL"
  uppercase={false}
>
  MONO
</Heading>
`;

exports[`renders mono dark gray Heading with default prop XL 1`] = `
.c0 {
  font-size: 40px;
  font-family: Source Code Pro,monospace;
  color: #16111C;
  margin-top: 50px;
  margin-bottom: 50px;
}

<Heading
  bold={false}
  className="c0"
  darkGray={true}
  marginBottom={null}
  marginTop={null}
  mono={true}
  size="XL"
  uppercase={false}
>
  DARK GRAY MONO
</Heading>
`;

exports[`renders purple Heading 1`] = `
.c0 {
  font-size: 40px;
  font-family: Lato,sans-serif;
  color: #6C5FC7;
  margin-top: 50px;
  margin-bottom: 50px;
}

<Heading
  bold={false}
  className="c0"
  marginBottom={null}
  marginTop={null}
  mono={false}
  purple={true}
  size="XL"
  uppercase={false}
>
  MONO
</Heading>
`;
